pyccel/ast/builtins.py

```diff
@@ -433,6 +434,64 @@ def __new__(cls, *args):
     def element(self):
         return self._args[0]
 
+#==============================================================================
+class PythonAbs(Function, PyccelAstNode):
+    """Represents a call to  python abs for code generation.
+
+    arg : Variable
+    """
+    def __init__(self, x):
+        self._shape     = x.shape
+        self._rank      = x.rank
+        self._dtype     = NativeInteger() if x.dtype is NativeInteger() else NativeReal()
+        self._precision = default_precision[str_dtype(self._dtype)]
+        self._order     = x.order
+
+    @property
+    def arg(self):
+        return self._args[0]
+
+#==============================================================================
+class PythonSum(Function, PyccelAstNode):
+    """Represents a call to  python sum for code generation.
+
+    arg : list , tuple , PythonTuple, Tuple, List, Variable
+    """
+
+    def __new__(cls, arg):
+        if not isinstance(arg, (list, tuple, PythonTuple, Tuple, List, Core.Variable, Expr)):
+            raise TypeError('Uknown type of  %s.' % type(arg))
+
+        return Basic.__new__(cls, arg)
+
+    def __init__(self, arg):
+        self._dtype = arg.dtype
+        self._rank  = 0
+        self._shape = ()
+        self._precision = default_precision[str_dtype(self._dtype)]
+
+    @property
+    def arg(self):
+        return self._args[0]
+
+
+    def fprint(self, printer, lhs=None):
+        """Fortran print."""
+
+        rhs_code = printer(self.arg)
+        if lhs:
+            lhs_code = printer(lhs)
+            return '{0} = sum({1})'.format(lhs_code, rhs_code)
+        return 'sum({0})'.format(rhs_code)
```

Comment:
Why do we need a `fprint` function here rather than using the standard method of creating a `_print_PythonSum` function in codegen/printing/fcode.py?

Reply:
I don't know why exactly. but I see you guys used it in many Numpy classes (see `numpyext.py`).

Reply:
In numpy it is usually done if the object can't be printed without a `lhs` (e.g. NumpyOnes).
Hopefully we can remove this separation after PR #521 goes through.
If you can't think of a reason then I would definitely move it to codegen/printing/fcode.py which is where we would normally search for it